### PR TITLE
Update grids.md

### PR DESCRIPTION
**Pull request by tomchor**
Added a list of available topologies

### CODE REVIEW

File: docs/src/model_setup/grids.md [hunk 1]
```diff
@@ -34,7 +34,16 @@ grid spacing (Δx, Δy, Δz): (4.0, 4.0, 2.0)
...
-`Flat` dimensions are used to specify two-dimensional and one-dimensional domains.
+`Flat` dimensions are used to specify two-dimensional and one-dimensional domains. 
+Currently only a subset of the possible topologies are available (the limiting factor being the pressure solver):
+
+```
+(Periodic, Periodic, Periodic)
+(Periodic, Periodic, Bounded)
+(Periodic, Bounded, Bounded)
+(Bounded, Bounded, Bounded) (only works on CPUs)
+```
```
Comment (glwagner) [inline docs/src/model_setup/grids.md:45]: A small edit could upgrade this block to bona fide julia code:

```julia
topology = (Periodic, Periodic, Periodic)
topology = (Periodic, Periodic, Bounded)
topology = (Periodic, Bounded, Bounded)
topology = (Bounded, Bounded, Bounded)
```

(Note that `topology = (Bounded, Bounded, Bounded)` only works on CPUs.)

I think it'd make the most sense to put this comment _after_ the example that shows how to specify a topology, perhaps?

**Review by glwagner — APPROVED** [review]
Looks great and thank you very much @tomchor. 

My comment suggests what could possibly be a small improvement to the clarity of the new addition. I also suggest moving the new text to the end of the subsection "Specifying the grid's topology".

File: docs/src/model_setup/grids.md (hunk 1 above)
Comment (tomchor) [reply docs/src/model_setup/grids.md:45]: Thanks! I agree that makes it clearer

Comment (glwagner) [reply docs/src/model_setup/grids.md:45]: I can take another look if you re-request a review once that's done.

**Comment by ali-ramadhan** [comment]
Awesome, thanks @tomchor!

Just a heads up that it might take a while for the docs to build and for your changes to appear in the documentation because of https://github.com/CliMA/Oceananigans.jl/issues/1186#issuecomment-732312213

**Comment by tomchor** [comment]
I tried to change the file to make some changes. Not sure if I did the right thing! (Sorry, first time contributing...)

File: docs/src/model_setup/grids.md [hunk 2]
```diff
@@ -34,7 +34,8 @@ grid spacing (Δx, Δy, Δz): (4.0, 4.0, 2.0)
 
 Another crucial keyword is a 3-`Tuple` that specifies the grid's `topology`.
 In each direction the grid may be `Periodic`, `Bounded`, or `Flat`.
```
Comment (ali-ramadhan) [inline docs/src/model_setup/grids.md:36]:
```suggestion
In each direction the grid may be `Periodic` or `Bounded`.
```

File: docs/src/model_setup/grids.md [hunk 3]
```diff
@@ -34,7 +34,8 @@ grid spacing (Δx, Δy, Δz): (4.0, 4.0, 2.0)
...
-`Flat` dimensions are used to specify two-dimensional and one-dimensional domains.
+`Flat` dimensions are used to specify two-dimensional and one-dimensional domains. 
+
```
Comment (ali-ramadhan) [inline docs/src/model_setup/grids.md:38]:
```suggestion
```

**Review by ali-ramadhan — APPROVED** [review]
I guess since we don't currently recommend using `Flat`, I made a couple of suggestions to your PR that remove any mentions of the `Flat` topology.

**Comment by glwagner** [comment]
@tomchor you can use the web interface to automatically accept @ali-ramadhan suggested changes:

image

File: docs/src/model_setup/grids.md [hunk 4]
```diff
@@ -51,6 +52,15 @@ RegularCartesianGrid{Float64, Periodic, Bounded, Bounded}
 grid spacing (Δx, Δy, Δz): (156.25, 156.25, 31.25)
 ```
 
+Note that currently only a subset of the possible topologies are available (the limiting factor being the pressure solver):
+
+```
+Topology = (Periodic, Periodic, Periodic)
+Topology = (Periodic, Periodic, Bounded)
+Topology = (Periodic, Bounded, Bounded)
+Topology = (Bounded, Bounded, Bounded) # Note: only works on CPUs
```
Comment (glwagner) [inline docs/src/model_setup/grids.md:61]: We prefer to use lowercase (snake-case) for variables, and reserve capitalized names for `struct`s and `module`s (roughly following https://github.com/jrevels/YASGuide). So these should all be `topology = ...`.

Comment (glwagner) [reply docs/src/model_setup/grids.md:61]: (and the comment is nice :-D)

**Review by navidcy — APPROVED** [review]
Great

**Pull request merged**